actionpack/test/dispatch/request_test.rb

```diff
@@ -23,14 +19,23 @@ def url_for(options = {})
   private
     def stub_request(env = {})
       ip_spoofing_check = env.key?(:ip_spoofing_check) ? env.delete(:ip_spoofing_check) : true
+      ActionDispatch::Http::URL.tld_length = env.delete(:tld_length) if env.key?(:tld_length)
+
+      uri = env["HTTPS"] == "on" ? "https://www.example.org" : "http://www.example.org"
```

Comment:
Explicitly defining the url based on `env["HTTPS"]` as `Rack::MockRequest` relies on the URI exclusively: https://github.com/rack/rack/blob/444dc8a130663159ed490f25948b73ff9b5fbba0/lib/rack/mock_request.rb#L110-L111

Reply:
Does our custom value not get set at the end of `env_for`?

https://github.com/rack/rack/blob/444dc8a130663159ed490f25948b73ff9b5fbba0/lib/rack/mock_request.rb#L155-L157

Reply:
It does set, but [RACK_URL_SCHEME](https://github.com/rack/rack/blob/main/lib/rack/constants.rb#L48) is not, causing the scheme to remain as http.

Without `uri = env["HTTPS"] == "on" ? "https://www.example.org" : "http://www.example.org"` the test errors with:
```
Failure:
RequestPort#test_standard_port? [/home/spin/src/github.com/rails/rails/actionpack/test/dispatch/request_test.rb:354]:
Expected #<ActionDispatch::Request GET "https://example.org:80/" for > to be standard_port?.
```

Here's the dump of `env` in mock_request for this test:
```
Stop by #0  BP - Line  /home/spin/.bundle/rails/gems/rack-3.0.8/lib/rack/mock_request.rb:163 (line)
(rdbg@bin/test#657085) env
{"rack.input"=>#<StringIO:0x00007f0461cd7848>,
 "rack.errors"=>#<StringIO:0x00007f0461cd7938>,
 "REQUEST_METHOD"=>"GET",
 "SERVER_NAME"=>"example.org",
 "SERVER_PORT"=>"80",
 "SERVER_PROTOCOL"=>"HTTP/1.1",
 "QUERY_STRING"=>"",
 "PATH_INFO"=>"/",
 "rack.url_scheme"=>"http",
 "HTTPS"=>"on",
 "SCRIPT_NAME"=>"",
 "CONTENT_LENGTH"=>"0"}
```

Here's the same structure for the same test, after setting the fix:
```
Stop by #0  BP - Line  /home/spin/.bundle/rails/gems/rack-3.0.8/lib/rack/mock_request.rb:163 (line)
(rdbg@bin/test#658127) env
{"rack.input"=>#<StringIO:0x00007fc6b73a4f00>,
 "rack.errors"=>#<StringIO:0x00007fc6b73a4ff0>,
 "REQUEST_METHOD"=>"GET",
 "SERVER_NAME"=>"www.example.org",
 "SERVER_PORT"=>"443",
 "SERVER_PROTOCOL"=>"HTTP/1.1",
 "QUERY_STRING"=>"",
 "PATH_INFO"=>"/",
 "rack.url_scheme"=>"https",
 "HTTPS"=>"on",
 "SCRIPT_NAME"=>"",
 "CONTENT_LENGTH"=>"0"}
```